### ISSUE TITLE
Training on Arabic language

### ISSUE THREAD

**Issue by njiekang** [bgshih/crnn#130]
I have generated data.mdb with images that contain a Arabic word in each of them.

I've made the following modification in create_dataset.py. That is, I've encoded the labels into utf-8 to avoid ascii error.

```
 def writeCache(env, cache):
    with env.begin(write=True) as txn:
        for k, v in cache.iteritems():
            #print(k ,v)
            if 'label' in k:
              txn.put(k.encode(),v.encode('utf-8'))
            else:
              txn.put(k.encode(),v)
```

I am not sure how to deal with str2label() and ascii2label() in [utilities.lua](https://github.com/bgshih/crnn/blob/master/src/utilities.lua)

In str2label(), 

```
for i, str in ipairs(strs) do
        for j = 1, string.len(str) do
            local ascii = string.byte(str, j)          -- this segments by 'half a character' due to the utf-8
            labels[i][j] = ascii2label(ascii)
        end
    end
```
 To illustrate what I mean by 'half a character',
original word: 'ستشهاديتان'
after encode('utf-8'): b'\xd8\xb3\xd8\xaa\xd8\xb4\xd9\x87\xd8\xa7\xd8\xaf\xd9\x8a\xd8\xaa\xd8\xa7\xd9'
string.byte() will take \xd8 as one character, but a character is actually made up of \xd8\xb3
Correction: I just realise that a character may use up to 3 byte

**Comment by githubharald** [comment]
the neural network is in fact only interested in labels representing chars (0, 1, 2, ...).
Why not simply map each of your arabic chars (UTF8, containing 1 or multiple bytes)  to a number in 0..255 (byte)? It should of course be a continuous range of numbers, so if you have 5 chars, then assign the numbers 0, 1, 2, 3, 4.
Then you can map your arabic string to a byte string in which each char uses only 1 byte.
The neural networks outputs a byte string containing the labels, which you can map back to a arabic text.

**Comment by njiekang** [comment]
@githubharald Thanks! I tried to do that.

Could anyone advise how to iterate through the utf-8 literal properly and apply string.byte( ) in lua?

**Comment by githubharald** [comment]
no, sorry.
If you can't find a solution for LUA, you could consider switching to a Python/TensorFlow implementation of CRNN (just search for it on github, there should be multiple repos). In Python this task should be pretty easy.
